src/tokens/tokens.service.ts

```diff
@@ -122,15 +109,10 @@ export class TokensService {
     }
   }
 
-  private async getStream(ctx: Context) {
-    const stream = this.stream;
-    if (stream !== undefined) {
-      return stream;
-    }
+  private async getStream(ctx: Context, namespace: string) {
     await this.migrationCheck(ctx);
     this.logger.log('Creating stream with name ' + this.topic);
-    this.stream = await this.eventstream.createOrUpdateStream(ctx, this.topic, this.topic);
-    return this.stream;
+    return this.eventstream.createOrUpdateStream(ctx, `${this.topic}/${namespace}`, this.topic);
```

Comment:
There was a caching aspect to this method before, which we've lost. Are we going to `PUT` the stream a lot more often now? Or does it only happen when we're initializing a namespace/pool?

Reply:
This may be worth a quick conversation. I'm not sure what we've lost, but I'm happy to talk through how it works now.